test/format/ops.c

```diff
@@ -523,7 +524,14 @@ begin_transaction(TINFO *tinfo, WT_SESSION *session, u_int *iso_configp)
 	}
 	*iso_configp = v;
 
-	testutil_check(session->begin_transaction(session, config));
+	/*
+	 * Keep trying to start a new transaction if it's timing out - we
+	 * know there aren't any resources pinned so it should succeed
+	 * eventually.
+	 */
+	while ((ret = session->begin_transaction(
+	    session, config)) == WT_CACHE_FULL) {}
+	testutil_check(ret);
```

Comment:
The alternative would be to not return an error from begin_transaction - but that would hobble the concept. It's nice to know that threads will *never* wait forever..

Reply:
Same KNF.

Reply:
I agree `session.begin_transaction` should return the error.